### ISSUE TITLE
CI failure due to dependency conflict

### ISSUE THREAD

**Issue by fjetter** [dask/distributed#8614]
All our CI jobs with python>3.10 are failing due to a dependency resolution conflict

```
The conflict is caused by:
    The user requested fsspec 2024.3.2.dev5+g2dd9355 (from git+https://github.com/fsspec/filesystem_spec)
    dask 2024.4.1+4.gb2ec1e1a depends on fsspec>=2021.09.0
    s3fs 2024.3.1+2.g7ccc0a5 depends on fsspec==2024.3.1
```

cc @martindurant are you aware of anything that caused this hard pinning between s3fs and fsspec?

e.g. https://github.com/dask/distributed/actions/runs/8619396892/job/23704420478?pr=8610

**Comment by martindurant** [comment]
Yes, s3fs (and gcsfs) are hard pinned to exactly the same version as fsspec, and have been for a long time. They are always released within hours of each other.

The latest release is 2024.3.1. I am thinking that the introduction of hatch in https://github.com/fsspec/filesystem_spec/pull/1553 has caused the dev version to be called "2024.3.2.XXX" instead of "2024.3.1.XXX" as versioneer used to do. I am not sure how to change this convention.

**Comment by fjetter** [comment]
OK, I don't want our entire CI to be blocked by this and opened https://github.com/dask/distributed/pull/8615 to temporarily just run CI against stable versions until this is figured out

**Comment by phofl** [comment]
The pin itself seems fine, I am a bit puzzled where the tag 2024.3.2 from fsspec is coming from, I can't find that tag in the repository.

**Comment by martindurant** [comment]
https://github.com/fsspec/filesystem_spec/pull/1569 should fix this, if you care to try

**Comment by phofl** [comment]
#8618 tries this

**Comment by martindurant** [comment]
Please try again with main/master whenever you think the caches might have cleared.

**Comment by phofl** [comment]
did that on my revert pr
